sympy/ntheory/primetest.py

```diff
@@ -110,6 +110,31 @@ def mr(n, bases):
     return True
 
 
+def isprimeE(n, b):
+    """Test whether n is prime or an Euler pseudoprime to base b."""
+	if (not mr(n, [b])):
+        return False
+    
+    r = int(r)
+	r = n-1
+	while (r % 2 == 0):
+        r //= 2
+    
+	c = pow(b, r, n)
+	if (c == 1):
+        return True
+    
+	while (1):
```

Comment:
Use `while True`.

Reply:
Yeah technically while True is more correct,
Thanks for the suggestion.